geopyspark/geotrellis/protobufcodecs.py

```diff
@@ -27,28 +28,28 @@
 # DECODERS
 
 
-def from_pb_tile(tile, data_type=None):
+def from_pb_tile(tile, no_data_value=None, data_type=None):
     if not data_type:
         data_type = _mapped_data_types[tile.cellType.dataType]
 
     if data_type == 'BIT':
-        cells = np.int8(tile.uint32Cells[:]).reshape(tile.rows, tile.cols)
+        cells = np.int8([no_data_value if x == SCALA_MAX_INT else x for x in tile.uint32Cells])
```

Comment:
It should be possible to avoid these if during encoding the tiles are converted to "raw" form of their cell type, which will expose the NoData values as the actual values rather than `Int.MinValue` and `Double.NaN`

here is some proof:
```scala
cala> val tile = ArrayTile.empty(ShortConstantNoDataCellType, 4,4)
tile: geotrellis.raster.MutableArrayTile = ShortConstantNoDataArrayTile([S@2091c7da,4,4)

scala> tile.toArray()
res6: Array[Int] = Array(-2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648, -2147483648)

scala> tile.interpretAs(ShortCellType).toArray()
res7: Array[Int] = Array(-32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768, -32768)
```